backend/homework/1/data/src/main/java/StudentRepo.java

```diff
@@ -0,0 +1,48 @@
+import java.util.*;
+//import org.backend1.*;
+public class StudentRepo {
+
+//    multiple student objects
+    private Student[] st;
+    private int index;
+
+    public StudentRepo(int size){
+        st = new Student[size];
+        index = 0;
+    }
+
+    // methods to add, update and retreive
+
+    public void addStudent(Student obj)
+    {
+        // checking repo size
+        if(index < st.length - 1){
+            st[index++] = obj;
+        }else System.out.println("Repository Full");
+    }
+
+    public void updateStudent(Student obj, int idx)
+    {
+        if(idx < st.length - 1) st[idx] = obj;
+        else System.out.println("Update not Possible, out of bounds");
+    }
+
+    public Student retrieveDetails(int id){
+        for (Student student : st) {
+            if (student != null && student.getId() == id) {
+                return student;
+            }
+        }
+        return null;
+    }
+
+    public Student retrieveDetails(String name){
```

Comment:
**Common comment** 
Always describe in single line what is the functionality of the defined function